packages/jaeger-ui/src/model/link-patterns.js

```diff
@@ -0,0 +1,183 @@
+    const text = processTemplate(pattern.text, identity);
+    return {
+      object: pattern,
+      type: createTestFunction(pattern.type),
+      key: createTestFunction(pattern.key),
+      value: createTestFunction(pattern.value),
+      url,
+      text,
+      parameters: _uniq(url.parameters.concat(text.parameters)),
+    };
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(`Ignoring invalid link pattern: ${error}`, pattern);
+    return null;
+  }
+}
+
+export function getParameterInArray(name, array) {
+  if (array) {
+    return array.find(entry => entry.key === name);
+  }
+  return undefined;
+}
+
+export function getParameterInAncestor(name, spans, startSpanIndex) {
+  let currentSpan = { depth: spans[startSpanIndex].depth + 1 };
+  for (let spanIndex = startSpanIndex; spanIndex >= 0; spanIndex--) {
+    const nextSpan = spans[spanIndex];
+    if (nextSpan.depth < currentSpan.depth) {
+      currentSpan = nextSpan;
+      const result =
+        getParameterInArray(name, currentSpan.tags) || getParameterInArray(name, currentSpan.process.tags);
+      if (result) {
+        return result;
+      }
+    }
+  }
+  return undefined;
+}
+
+export function callTemplate(template, data) {
+  return template.template(...template.parameters.map(param => data[param]));
+}
+
+export function computeLinks(linkPatterns, trace, spanIndex, items, itemIndex) {
+  const item = items[itemIndex];
+  const span = trace.spans[spanIndex];
+  let type = 'logs';
+  const processTags = span.process.tags === items;
+  if (processTags) {
+    type = 'process';
+  }
+  const spanTags = span.tags === items;
+  if (spanTags) {
+    type = 'tags';
+  }
+  const result = [];
+  linkPatterns.forEach(pattern => {
+    if (pattern.type(type) && pattern.key(item.key, item.value, type) && pattern.value(item.value)) {
+      let parameterValues = {};
+      pattern.parameters.every(parameter => {
+        let entry = getParameterInArray(parameter, items);
+        if (!entry && !processTags) {
+          // do not look in ancestors for process tags because the same object may appear in different places in the hierarchy
+          entry = getParameterInAncestor(parameter, trace.spans, spanIndex);
+        }
+        if (entry) {
+          parameterValues[parameter] = entry.value;
+          return true;
+        }
+        // eslint-disable-next-line no-console
+        console.warn(
+          `Skipping link pattern, missing parameter ${parameter} for key ${item.key} in ${type}.`,
+          pattern.object
+        );
+        parameterValues = null;
+        return false;
+      });
+      if (parameterValues) {
+        result.push({
+          url: callTemplate(pattern.url, parameterValues),
+          text: callTemplate(pattern.text, parameterValues),
+        });
+      }
+    }
+  });
+  return result;
+}
+
+const linkPatterns = (getConfigValue('linkPatterns') || []).map(processLinkPattern).filter(value => !!value);
```

Comment:
This can be shortened to `.filter(Boolean)`.

Reply:
Thank you! I have updated the PR.